src/ParseFile.js

```diff
@@ -292,6 +300,52 @@ class ParseFile {
     );
   }
 
+  /**
+   * Sets metadata to be saved with file object. Overwrites existing metadata
+   * @param {Object} metadata Key value pairs to be stored with file object
+   */
+  setMetadata(metadata: Object) {
+    if (metadata && typeof metadata === 'object') {
+      Object.keys(metadata).forEach((key) => {
+        this.addMetadata(key, metadata[key]);
+      });
+    }
+  }
+
+  /**
+   * Sets metadata to be saved with file object. Adds to existing metadata
+   * @param {String} key
+   * @param {String} value
+   */
+  addMetadata(key: String, value: String) {
```

Comment:
This can be removed  and just use this._metadata = metadata?

Reply:
I added these so that you can add to existing meta data without having to get the existing and add to it

Reply:
Ok, I think that value should be generic `value: any`. The storage adapters should handle the validation if any.

Reply:
This has been updated